autorest/date/timeRfc1123_test.go

```diff
@@ -0,0 +1,196 @@
+
+	err := d.UnmarshalText([]byte(t))
+	if err != nil {
+		fmt.Println(err)
+	}
+	fmt.Println(d)
+	// Output: Sat, 03 Feb 2001 04:05:06 UTC
+}
+
+func TestUnmarshalJSONforInvalidDateRfc1123(t *testing.T) {
+	dt := `"Mon, 02 Jan 2000000 15:05 MST"`
+	d := TimeRfc1123{}
+	err := d.UnmarshalJSON([]byte(dt))
+	if err == nil {
+		t.Errorf("date: TimeRfc1123#Unmarshal failed for invalid date")
+	}
+}
+
+func TestUnmarshalTextforInvalidDateRfc1123(t *testing.T) {
+	dt := "Mon, 02 Jan 2000000 15:05 MST"
+	d := TimeRfc1123{}
+	err := d.UnmarshalText([]byte(dt))
+	if err == nil {
+		t.Errorf("date: TimeRfc1123#Unmarshal failed for invalid date")
+	}
+}
+
+func TestTimeStringRfc1123(t *testing.T) {
+	ti, _ := ParseTime(rfc1123, "Mon, 02 Jan 2006 15:04:05 MST")
+	d := TimeRfc1123{ti}
+	if d.String() != "Mon, 02 Jan 2006 15:04:05 MST" {
+		t.Errorf("date: TimeRfc1123#String failed (%v)", d.String())
+	}
+}
+
+func TestTimeStringReturnsEmptyStringForErrorRfc1123(t *testing.T) {
+	d := TimeRfc1123{Time: time.Date(20000, 01, 01, 01, 01, 01, 01, time.UTC)}
+	if d.String() != "" {
+		t.Errorf("date: TimeRfc1123#String failed empty string for an error")
+	}
+}
+
+func TestTimeBinaryRoundTripRfc1123(t *testing.T) {
+	ti, err := ParseTime(rfc3339, "2001-02-03T04:05:06Z")
+	d1 := TimeRfc1123{ti}
+	t1, err := d1.MarshalBinary()
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#MarshalBinary failed (%v)", err)
+	}
+
+	d2 := TimeRfc1123{}
+	err = d2.UnmarshalBinary(t1)
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#UnmarshalBinary failed (%v)", err)
+	}
+
+	if !reflect.DeepEqual(d1, d2) {
+		t.Errorf("date: Round-trip Binary failed (%v, %v)", d1, d2)
+	}
+}
+
+func TestTimeJSONRoundTripRfc1123(t *testing.T) {
+	type s struct {
+		Time TimeRfc1123 `json:"datetime"`
+	}
+	var err error
+	ti, err := ParseTime(rfc1123, "Mon, 02 Jan 2006 15:04:05 MST")
+	d1 := s{Time: TimeRfc1123{ti}}
+	j, err := json.Marshal(d1)
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#MarshalJSON failed (%v)", err)
+	}
+
+	d2 := s{}
+	err = json.Unmarshal(j, &d2)
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#UnmarshalJSON failed (%v)", err)
+	}
+
+	if !reflect.DeepEqual(d1, d2) {
+		t.Errorf("date: Round-trip JSON failed (%v, %v)", d1, d2)
+	}
+}
+
+func TestTimeTextRoundTripRfc1123(t *testing.T) {
+	ti, err := ParseTime(rfc1123, "Mon, 02 Jan 2006 15:04:05 MST")
+	d1 := TimeRfc1123{Time: ti}
+	t1, err := d1.MarshalText()
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#MarshalText failed (%v)", err)
+	}
+
+	d2 := TimeRfc1123{}
+	err = d2.UnmarshalText(t1)
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#UnmarshalText failed (%v)", err)
+	}
+
+	if !reflect.DeepEqual(d1, d2) {
+		t.Errorf("date: Round-trip Text failed (%v, %v)", d1, d2)
+	}
+}
+
+func TestTimeToTimeRfc1123(t *testing.T) {
+	ti, err := ParseTime(rfc1123, "Mon, 02 Jan 2006 15:04:05 MST")
+	d := TimeRfc1123{ti}
+	if err != nil {
+		t.Errorf("date: TimeRfc1123#ParseTime failed (%v)", err)
+	}
+	var v interface{} = d.ToTime()
```

Comment:
@garimakhulbe if you want  to test that, there's an easier test:

     var v time.Time = d.ToTime()

no need for runtime type assertion, this will fail in compile-time if you change return type of `ToTime`